pkg/stanza/fileconsumer/internal/reader/compressedreader.go

```diff
@@ -0,0 +1,147 @@
+package reader
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/header"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/scanner"
+	"go.uber.org/zap"
+	"golang.org/x/net/context"
+	"math"
+	"os"
+)
+
+type CompressedReader struct {
+	*Reader
+
+	buffer           *bytes.Buffer
+	lastReadFileSize int64
+	fromBeginning    bool
+}
+
+func (r *CompressedReader) ReadToEnd(ctx context.Context) {
+	newFileHandle, err := os.Open(r.file.Name())
```

Comment:
The embedded Reader should have a handle to the file. Can we use that?

Reply:
good catch, not sure why i created a new file handle here (seems to be a residue from a previous approach i tried out)